### ISSUE TITLE
Kucoin - Price queried from get_price_for_volume is not correct

### ISSUE THREAD

**Issue by riven314** [hummingbot/hummingbot#6244]
**Describe the bug**
`get_price_for_volume` queries the average execution price taking into account the order amount and current order book, so this price should be no less than the best ask if it is a buy, no more than the best bid if it is a sell.

But this condition is not met when I used `get_price_for_volume` in kucoin spot. Other exchanges like Binance are working fine with `get_price_for_volume`.

**Steps To Reproduce**
Run the following script:
```
from decimal import Decimal
from enum import Enum

from hummingbot.strategy.script_strategy_base import ScriptStrategyBase


class LogBidAskPrice(ScriptStrategyBase):
    spot_connector = "bybit"
    trading_pair = "ARB-USDT"
    # markets = {spot_connector: {trading_pair}, perp_connector: {trading_pair}}
    markets = {spot_connector: {trading_pair}}
    base_order_amount = Decimal("0.1")

    def on_tick(self):
        spot_best_bid_price = self.connectors[self.spot_connector].get_price(
            self.trading_pair, False
        )
        spot_best_ask_price = self.connectors[self.spot_connector].get_price(
            self.trading_pair, True
        )

        spot_taker_buy_price = self.limit_taker_price(self.spot_connector, True)
        spot_taker_sell_price = self.limit_taker_price(self.spot_connector, False)

        self.logger().info(f"Spot best bid price: {spot_best_bid_price}")
        self.logger().info(f"Spot best ask price: {spot_best_ask_price}")
        self.logger().info(f"Spot taker buy price: {spot_taker_buy_price}")
        self.logger().info(f"Spot taker sell price: {spot_taker_sell_price}")

    def limit_taker_price(self, connector_name: str, is_buy: bool) -> Decimal:
        limit_taker_price_result = self.connectors[connector_name].get_price_for_volume(
            self.trading_pair, is_buy, self.base_order_amount
        )
        return limit_taker_price_result.result_price
```

**Screenshots**
<img width="862" alt="Screenshot 2023-04-23 at 4 38 13 AM" src="https://user-images.githubusercontent.com/21143399/233805661-8c4f1c12-57fb-42f8-a7bd-ceef1f3c251f.png">

Capture part of the above logs here, so as to illustrate the issue:
```
2023-04-22 16:27:17,003 - 1 - hummingbot.strategy.script_strategy_base - INFO - Spot best bid price: 1.475
2023-04-22 16:27:17,008 - 1 - hummingbot.strategy.script_strategy_base - INFO - Spot best ask price: 1.476
2023-04-22 16:27:17,010 - 1 - hummingbot.strategy.script_strategy_base - INFO - Spot taker buy price: 1.475
2023-04-22 16:27:17,016 - 1 - hummingbot.strategy.script_strategy_base - INFO - Spot taker sell price: 1.475
```
As you can see, when best ask price is 1.476, the taker buy price should be 1.475, which doesn't make sense to me

**Release version**
1.14.0

**Attachments**
[logs_log_bid_ask_price.log](https://github.com/hummingbot/hummingbot/files/11302243/logs_log_bid_ask_price.log)

Optional: alex_lau#8125

**Comment by cardosofede** [comment]
@riven314 can you test with this PR: https://github.com/hummingbot/hummingbot/pull/6149

**Comment by riven314** [comment]
Hi @nikspz  @cardosofede !
I just tested the same script on https://github.com/hummingbot/hummingbot/pull/6149 and found that it fixes the issue
here is the logs when I deployed the PR. As you can see, taker buy price >= best ask AND taker sell price <= best bid
![Screenshot 2023-05-31 at 2 25 00 AM](https://github.com/hummingbot/hummingbot/assets/21143399/72da3962-b8aa-419d-9ae2-e9dad9f834eb)

**Comment by nikspz** [comment]
Fixed with #6149, merged to dev-1.17.0 and will be available in 1.17.0 Release
